File: tmp_upload_check.mjs
import { chromium } from "playwright";

const FRONTEND_URL = "http://localhost:5173";
const XML_PATH = "C:/Users/Oem/OneDrive/Desktop/saas-fiscal-demo/app/xmls_testes/xml_icms10_st_real.xml";

const logs = [];
let uiMessage = null;

const browser = await chromium.launch({
  headless: true,
  channel: "msedge"
});

const page = await browser.newPage();
page.on("console", (msg) => {
  logs.push(msg.text());
});

await page.goto(FRONTEND_URL, { waitUntil: "domcontentloaded" });

await page.fill('input[placeholder="email"]', "demo@demo.com");
await page.fill('input[placeholder="senha"]', "qualquer");
await page.click('button[type="submit"]');

await page.waitForSelector('input[type="file"]', { timeout: 15000 });
await page.setInputFiles('input[type="file"]', XML_PATH);

await page.waitForTimeout(5000);

const jobLine = logs.find((line) => line.includes("JOB:"));
if (jobLine) {
  console.log(`RESULT_JOB_LOG=${jobLine}`);
} else {
  const bodyText = await page.textContent("body");
  uiMessage = bodyText
    ?.split("\n")
    .map((x) => x.trim())
    .filter(Boolean)
    .find((x) => x.toLowerCase().includes("erro") || x.toLowerCase().includes("upload") || x.toLowerCase().includes("xml"));
  if (uiMessage) {
    console.log(`RESULT_UI=${uiMessage}`);
  } else {
    console.log("RESULT_NONE=Sem JOB no console e sem mensagem visível clara.");
  }
}

await browser.close();

File: tmp_upload_check.py
from playwright.sync_api import sync_playwright

FRONTEND_URL = "http://localhost:5173"
XML_PATH = r"C:\Users\Oem\OneDrive\Desktop\saas-fiscal-demo\app\xmls_testes\xml_icms10_st_real.xml"

logs = []
ui_message = None

with sync_playwright() as p:
    browser = p.chromium.launch(headless=True, channel="msedge")
    page = browser.new_page()

    page.on("console", lambda msg: logs.append(msg.text))

    page.goto(FRONTEND_URL, wait_until="domcontentloaded")
    page.fill('input[placeholder="email"]', "demo@demo.com")
    page.fill('input[placeholder="senha"]', "qualquer")
    page.click('button[type="submit"]')

    page.wait_for_selector('input[type="file"]', timeout=15000)
    page.set_input_files('input[type="file"]', XML_PATH)
    page.wait_for_timeout(6000)

    job_line = next((line for line in logs if "JOB:" in line), None)
    if job_line:
        print(f"RESULT_JOB_LOG={job_line}")
    else:
        body_text = page.text_content("body") or ""
        candidates = [x.strip() for x in body_text.splitlines() if x.strip()]
        ui_message = next(
            (
                x
                for x in candidates
                if "erro" in x.lower() or "upload" in x.lower() or "xml" in x.lower()
            ),
            None,
        )
        if ui_message:
            print(f"RESULT_UI={ui_message}")
        else:
            print("RESULT_NONE=Sem JOB no console e sem mensagem visível clara.")

    browser.close()
